libnavui-app/src/main/java/com/mapbox/navigation/ui/app/internal/controller/StateResetController.kt

```diff
@@ -0,0 +1,37 @@
+package com.mapbox.navigation.ui.app.internal.controller
+
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.internal.extensions.flowTripSessionState
+import com.mapbox.navigation.core.trip.session.TripSessionState
+import com.mapbox.navigation.ui.app.internal.Store
+import com.mapbox.navigation.ui.base.lifecycle.UIComponent
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.flow.debounce
+
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class, FlowPreview::class)
+internal class StateResetController(private val store: Store) : UIComponent() {
+
+    override fun onAttached(mapboxNavigation: MapboxNavigation) {
+        super.onAttached(mapboxNavigation)
+
+        // TripSessionState might change in quick succession.
```

Comment:
Might? or does the debounce fix an issue you found while testing?

Reply:
TripSessionState changes in quick succession only when the replayer is enabled/disabled. Since replayer usage is optional ---> "might".

Reply:
Can you update the comment to be specific?
```suggestion
        // TripSessionState is triggered rapidly when enabling and disabling replay when calling `MapboxNavigation`; `startTripSession, `stopTripSession`, `startReplayTripSession`
```

I'm thinking that is an issue with `startReplayTripSession`, and we may be able to remove the debounce if we make it so calling `startReplayTripSession` does not require a `MapboxNavigation.stopTripSession()` call.

`startReplayTripSession` is [experimental for this reason](https://github.com/mapbox/mapbox-navigation-android/blob/9f849f7f3051c85cfc6808474e8734ae1b482931/libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt#L594-L597), we may want to update how it is used.

Reply:
Correct. Once we patch how `startReplayTripSession` works, we'll remove the debounce.